File: python-algorithm/ds_5_tree/tree_1_4_path_recurrent/L404.py
#--coding:utf-8--
'''
@Time   : 2020/10/15
@Author : No Name
'''

class Solution:
    res = 0

    def sumOfLeftLeaves(self, root: TreeNode) -> int:
        if not root:
            return 0
        self.find_left_leaf(root, root)
        return self.res

    def find_left_leaf(self, cur, pre):
        if not cur.left and not cur.right and pre.left == cur:
            self.res += cur.val
        if cur.left:
            self.find_left_leaf(cur.left, cur)
        if cur.right:
            self.find_left_leaf(cur.right, cur)

File: python-algorithm/ds_5_tree/tree_1_4_path_recurrent/L404.cpp

//左叶子节点的和

#include <iostream>
#include <vector>
using namespace std;

// Definition for a binary tree node.
struct TreeNode {
    int val;
    TreeNode *left;
    TreeNode *right;
    TreeNode(int x) : val(x), left(NULL), right(NULL) {}
};

//递归，前序遍历
class Solution {
public:
    vector<int> temp;
    void dfs(TreeNode* root,TreeNode* father){
        if(father->left == root && !root->left && !root->right)
            temp.push_back(root->val);
        if(root->left != NULL)
            dfs(root->left, root);
        if(root->right != NULL)
            dfs(root->right, root);
    }

    int sumOfLeftLeaves(TreeNode* root) {
        int res = 0;
        if(root == NULL) return 0;
        dfs(root, root);
        for(auto item:temp)
            res += item;
        return res;
    }
};